crypto/objects/obj_dat.c

```diff
@@ -790,6 +790,10 @@ int OBJ_create(const char *oid, const char *sn, const char *ln)
     } else {
         /* Create a no-OID ASN1_OBJECT */
         tmpoid = ASN1_OBJECT_new();
+        if (tmpoid == NULL) {
+            ERR_raise(ERR_LIB_OBJ, ERR_R_ASN1_LIB);
```

Comment:
The `new` method failure is usually related to memory allocation failure. So it seems the reason should be either `ERR_R_MALLOC_FAILURE` or `ERR_R_INTERNAL_ERROR`?

Reply:
`ERR_R_ASN1_LIB` seems too high-level to find out what really happens

Reply:
I kind of disagree. The underlying CRYPTO_malloc() call will raise `ERR_R_MALLOC_FAILURE` if it is indeed a malloc failure, and at this level we do not know the reason for the failure - just that it failed - so this looks reasonable to me.